### DISCUSSION TITLE
Umi Max 为什么要依赖 antd？

### DISCUSSION THREAD

**Discussion by lanvige** [umijs/umi#10811] [Q&A]
在 Max 的主项目中直接依赖了 Antd 4 的版本。

https://github.com/umijs/umi/blob/master/packages/max/package.json

``` js
"dependencies": {
    "@umijs/lint": "workspace:*",
    "@umijs/plugins": "workspace:*",
    "antd": "^4.20.6",
    "eslint": "8.35.0",
    "stylelint": "14.8.2",
    "umi": "workspace:*"
},
```
 

而在 @umijs/plugins 中通过 devDependencies 依赖 antd 4。

``` js
"devDependencies": {
  "antd": "^4.24.1",
  "umi": "workspace:*"
},
```

第二种是个人理解上比较正确的，项目可以自己指定符合版本的 antd，比如说 v5。

但第一种是出于什么原因的呢？

**Comment by fz6m** [comment]
内置 antd 一个版本，用户不安装 antd 也可以用，如果需要自己用自己的版本，自己装了就会优先用你装的。

**Reply by lanvige** [reply]
> 内置 antd 一个版本，用户不安装 antd 也可以用，如果需要自己用自己的版本，自己装了就会优先用你装的。

我能理解您的意思，但从设计原则上，这样就依赖两个版本了？

同时，如果 antd v4 / v5 在 API 上有变化，就不能兼容不同版本了。

个人理解应该是 Max 并不依赖具体的 antd 版本，而 plugin/antd 会有针对不同版本的插件（在 antd API 有变更时）。

请问我这样合理吗？

**Reply by fz6m** [reply]
这种情况就算安装了两个 antd 依赖，是没关系的，因为插件里做了 alias 唯一化，至于你说的 antd 4 / 5 配置区别已经在插件里检测抹平了。
